containers/daap-python-base/src/var/task/data_platform_paths.py

```diff
@@ -108,6 +108,25 @@ def get_account_id() -> str:
     """
     return boto3.client("sts").get_caller_identity()["Account"]
 
+def search_string_for_regex(string: str, regex: str) -> str:
+    """Search a string for a regex pattern and return the first result"""
+    database_name_search = re.search(regex, string)
+    if database_name_search:
+        return database_name_search.groups()[0]
+    else:
+        raise ValueError(f"{regex} not found in {string}")
+
+
+def database_name_regex() -> str:
```

Comment:
Rather than exposing the regex itself, could we have functions for each one, like 

- `extract_timestamp_from_curated_path(string: str)`
- `extract_table_name_from_curated_path(string: str)`
- `extract_database_name_from_curated_path(string: str)`

it should simplify the calling code